interceptor.go

```diff
@@ -459,16 +459,21 @@ func fetch(this js.Value, args []js.Value) interface{} {
 				// Converting the body to Golag or setting it as null/nil
 				bodyMap := map[string]interface{}{}
 				body := options.Get("body")
+				urlPath := strings.Replace(spURL, host, "", 1)
+				// fmt.Println("[Interceptor] URL Path: ", urlPath) // For debugging purposes
 				if body.String() == "<undefined>" {
 					// body = js.ValueOf(map[string]interface{}{}) <= this will err out as "Uncaught (in promise) Error: invalid character '<' looking for beginning of value"
 					body = js.ValueOf("{}")
+					bodyMap["url_path"] = urlPath
 				} else {
 					err := json.Unmarshal([]byte(body.String()), &bodyMap)
 					if err != nil {
 						reject.Invoke(js.Global().Get("Error").New(err.Error()))
 						return
 					}
+					bodyMap["url_path"] = urlPath
```

Comment:
You can add the url_path when initializing `bodyMap` without having to repeat `bodyMap["url_path"] = urlPath` in the if...else.

Also since you've prefixed the url_path with double underscores, the same can be done here as well

```suggestion
				urlPath := strings.Replace(spURL, host, "", 1)
				bodyMap := map[string]interface{}{
				    "__url_path": urlPath,
				}
				body := options.Get("body")
				// fmt.Println("[Interceptor] URL Path: ", urlPath) // For debugging purposes
				if body.String() == "<undefined>" {
					// body = js.ValueOf(map[string]interface{}{}) <= this will err out as "Uncaught (in promise) Error: invalid character '<' looking for beginning of value"
					body = js.ValueOf("{}")
				} else {
					err := json.Unmarshal([]byte(body.String()), &bodyMap)
					if err != nil {
						reject.Invoke(js.Global().Get("Error").New(err.Error()))
						return
					}
```

Reply:
Agreed. I noticed the repeated code as well.
(Though I would be lying if I said I knew any solution for you. )